File: training/ruby/lib/de/sots/cellars_and_centaurs/domain/model/speed_type.rb
# frozen_string_literal: true

require_relative '../../version'

module De::Sots::CellarsAndCentaurs::Domain::Model
  module SpeedType
    WALKING = :walking
    FLYING = :flying
    SWIMMING = :swimming
    BURROWING = :burrowing
    CLIMBING = :climbing

    ALL = [WALKING, FLYING, SWIMMING, BURROWING, CLIMBING].freeze
  end
end

File: sample-projects/php/src/Domain/Model/SpeedType.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Domain\Model;

enum SpeedType: string
{
    case WALKING = 'walking';
    case FLYING = 'flying';
    case SWIMMING = 'swimming';
    case BURROWING = 'burrowing';
    case CLIMBING = 'climbing';
}
